src/pages/profiles/ProfileUsedByProject.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { FC } from "react";
+import InstanceLink from "pages/instances/InstanceLink";
+import { UsedByProject } from "util/usedBy";
+import ExpandableList from "components/ExpandableList";
+
+interface Props {
+  usedByProjObj: UsedByProject;
+}
+
+const ProfileUsedByProject: FC<Props> = ({ usedByProjObj }) => {
```

Comment:
Ok about removing the `UsedByProject` type, but moving the filtering logic to the child component is increasing complexity rather than reducing it. Passing the profile to the child component is not enough. It needs also the `usedByInstances` list to do the filtering. Also, there's this other thing to take into account: that the default profile should always be the first one.

Reply:
Right, I see what you mean. So maybe we should just flatten the structure of the interface. So it receives a `project` as string and a `instances` list?

What I found a bit odd atm are two things.
- We calculate the `otherProjects` and then the `usedByProject` in the `ProfileDetailOverview`. We do that for any project, even though we need it only for the default one.
- We display the default project in the list of used by, even if it has 0 entries - but we don't do it for the other projects.

I wonder if we should instead of calculating the `otherProjects` and then the `usedByProject` produce a list of projects: default project first, then other projects inheriting the profile. So we can use that list of project strings to create all the `ProfileUsedByProject` instances.

Reply:
Sorry, I removed my previous comment, and implemented your suggestion instead. Now the whole content of the "Instances" section is inside its dedicated component, freeing the `ProfileDetailOverview` of all that logic.

Reply:
That's already way better, because it contains the logic. What do you think about the later idea above
> produce a list of projects: default project first, then other projects inheriting the profile
So we always display all affected projects.

Reply:
> * We calculate the `otherProjects` and then the `usedByProject` in the `ProfileDetailOverview`. We do that for any project, even though we need it only for the default one.

No, we don't. `otherProjects` gets set to `[]` if the project is not the default one.

> * We display the default project in the list of used by, even if it has 0 entries - but we don't do it for the other projects.

I cannot really reproduce this case of 0 entries being displayed in any way, but I will try to make sure that the default project is not added when it has 0 instance entries.
Then, I will also set the `usedByProjectList` to be an empty array when the project is not the default one.

Reply:
> That's already way better, because it contains the logic. What do you think about the later idea above
> 
> > produce a list of projects: default project first, then other projects inheriting the profile
> 
> So we always display all affected projects.

I'm not really sure I understand this and what improvement it would make?

Reply:
> No, we don't. `otherProjects` gets set to `[]` if the project is not the default one.

Kind of what I meant. We set it to empty array, even though we don't need it.

> I'm not really sure I understand this and what improvement it would make?

The idea is to make it visible which projects that profile is affecting. ATM a profile will only pop up, if it has instances in it.

Reply:
> Kind of what I meant.

Not really - you said that we calculate it, but we aren't really calculating anything for that case, just setting an empty array. And I don't really see how that empty array can do any harm, I'm pretty sure we do similar things everywhere, so I don't really see anything (too) bad about it.

> The idea is to make it visible which projects that profile is affecting. ATM a profile will only pop up, if it has instances in it.

The project is affected by a profile if an instance belonging to that project has that profile assigned. And it will be correctly shown in that case, no? Do you have a concrete example of the current implementation not fulfilling the design requirement?